yt/utilities/logger.py

```diff
@@ -84,6 +84,28 @@ def filter(self, record):
 ytLogger.addFilter(DuplicateFilter())
 
 
+class DeprecatedFieldFilter(logging.Filter):
+    """A filter that suppresses repeated logging of deprecated field warnings"""
+
+    def __init__(self, name=""):
+        self.logged_fields = []
+        super().__init__(name=name)
+
+    def filter(self, record):
+        if record.msg.startswith("The Derived Field"):
+            field = record.args[0]
+            if field in self.logged_fields:
+                return False
+            else:
+                self.logged_fields.append(field)
+                return True
+        else:
+            return True
```

Comment:
Here's a flattened version of this logic tree
```suggestion
        if not record.msg.startswith("The Derived Field"):
            return True
            
        field = record.args[0]
        if field in self.logged_fields:
            return False

        self.logged_fields.append(field)
        return True
```